models/gamesM.js

```diff
@@ -1,25 +1,131 @@
-      LEFT JOIN comments
-      ON reviews.review_id = comments.review_id
-      GROUP BY reviews.review_id
-      ORDER BY reviews.created_at DESC
-      ;`
-    )
-    .then((result) => {
-      return result.rows;
+// exports.selectReviews = (category, sort_by, order) => {
+//   return pool
+//     .query(
+//       `SELECT
+
+//       title,
+//       category,
+//       designer,
+//       owner,
+//       review_img_url,
+//       reviews.created_at,
+//       reviews.review_id,
+//       reviews.votes,
+//       CAST(COUNT(comments.comment_id) AS INTEGER) as comment_count FROM reviews
+
+//       LEFT JOIN comments
+
+//         ON reviews.review_id = comments.review_id
+
+//       ${category ? `WHERE reviews.category=$1` : ``}
+//       GROUP BY reviews.review_id
+//       ORDER BY reviews.${sort_by} ${order}
+//     ;`,
+//       category ? [category] : []
+//     )
+//     .then((result) => {
+//       if (result.rows.length > 0) {
+//         return result.rows;
+//       } else if (category) {
+//         return [];
+//       } else {
+//         throw { status: 404, msg: "No reviews found" };
+//       }
+//     });
+// };
+
+exports.selectReviews = (sort_by = "created_at", order = "desc", category) => {
+  const validSortBy = [
+    "created_at",
+    "owner",
+    "category",
+    "review_id",
+    "votes",
+    "comment_count",
+    "title",
+    "designer",
+  ];
+
+  let querySoFar = `SELECT 
+
+  title, 
+  category, 
+  designer, 
+  owner, 
+  review_img_url, 
+  reviews.created_at, 
+  reviews.review_id, 
+  reviews.votes, 
+  CAST(COUNT(comments.comment_id) AS INTEGER) as comment_count FROM reviews
+
+  LEFT JOIN comments 
+  ON reviews.review_id = comments.review_id`;
+
+  let queryValues = [];
+  if (category) {
+    querySoFar += ` WHERE category = $1`;
+    queryValues.push(category);
+  }
+
+  querySoFar += ` GROUP BY reviews.review_id`;
+
+  if (!validSortBy.includes(sort_by)) {
+    return Promise.reject({
+      status: 400,
+      msg: "Invalid sort by",
+    });
+  }
+
+  if (sort_by === "review_id") {
+    sort_by = "reviews.review_id";
+  }
+  if (sort_by === "created_at") {
+    sort_by = "reviews.created_at";
+  }
+  if (sort_by === "votes") {
+    sort_by = "reviews.votes";
+  }
+  if (sort_by === "title") {
+    sort_by = "title::bytea";
+  }
+  if (sort_by === "designer") {
+    sort_by = "designer::bytea";
+  }
+  if (sort_by === "review_body") {
+    sort_by = "title::bytea";
+  }
+
+  if (order.toLowerCase() === "asc") {
+    querySoFar += ` ORDER BY ${sort_by} ASC`;
+  } else if (order.toLowerCase() === "desc") {
+    querySoFar += ` ORDER BY ${sort_by} DESC`;
+  } else {
+    return Promise.reject({
+      status: 400,
+      msg: "Invalid order",
+    });
+  }
+
+  querySoFar += `;`;
+ return checkCategoryExists(category).then(() => {
+
+    return pool.query(querySoFar, queryValues).then((reviews) => {
+      return reviews.rows;
```

Comment:
Looks like you have a nested `.then` here - I think your `then` block should go _after_ the closing bracket on like **128** rather than chained directly onto the end of the `pool.query()`